server/src/main/java/io/crate/metadata/doc/DocTableInfoFactory.java

```diff
@@ -242,6 +244,15 @@ public DocTableInfo create(RelationName relation, Metadata metadata) {
             mappingSource = Maps.getOrDefault(mappingSource, "default", mappingSource);
             tableParameters = indexTemplateMetadata.settings();
             versionCreated = tableParameters.getAsVersion(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT);
+            // Versions up to 5.9.7 had a bug where ALTER TABLE on a partitioned table could change the `versionCreated` property on the template.
+            // See https://github.com/crate/crate/pull/17178
+            // To mitigate the impact, this looks through partitions and takes their lowest version:
+            for (String indexName : concreteIndices) {
+                IndexMetadata indexMetadata = metadata.index(indexName);
+                if (indexMetadata != null) {
+                    versionCreated = Version.min(versionCreated, indexMetadata.getCreationVersion());
+                }
```

Comment:
AFAIK this issue is only partitioned table specific, so no need to fix regular tables?

Reply:
The code here is specific to partitioned tables?

Reply:
Ah right, got tricked by if (indexTemplateMetadata != null) and thought we are doing it in regular table branch.

We can drop if (indexTemplateMetadata != null) in the condition below (we are in else branch already anyway)

Reply:
> We can drop if (indexTemplateMetadata != null) in the condition below

We can't, the below block is outside of the first if because it needs access to the `references`.